src/core/fonts.js

```diff
@@ -2117,6 +2117,11 @@ class Font {
         break;
       }
 
+      // If the last offset is 0 in the loca table then we can't compute the
+      // endOffset for the last glyph. So in such a case we set the endOffset
+      // to the end of the data (fixes issue #17671).
+      locaEntries.at(-2).endOffset ||= oldGlyfDataLength;
+
```

Comment:
This seems somewhat related to existing code above, see https://github.com/mozilla/pdf.js/blob/14874e50b788cb8d2c57cdfa97d37fc0afe6c543/src/core/fonts.js#L2097-L2102

So, should this new block perhaps be placed just after (or somehow combined with) that code instead?

Reply:
Before running this loop, the locaEntries array is sorted by the offset when the fix is for the last element (last in term of index).
With the pdf in the bug, entry 10 (offset 0) is moved after entry 0 (offset 0 either), so the last element is now the offset for glyph 9. But for a non-buggy font, the last entry would be at the end because its offset is correct.
So I think it's the right place but I'm not 100% sure.